Prox/Prox/Data/PlacesController.swift

```diff
@@ -110,13 +117,36 @@ class PlacesProvider {
         }
     }
 
-    private func displayPlaces(places: [Place], forLocation location: CLLocation) {
-        let preparedPlaces = self.preparePlaces(places: places, forLocation: location)
+    private func fetchPlacesWithEvents(location: CLLocation) {
+        let eventProvider = EventsProvider()
+        eventsPlaces.removeValue(forKey: location)
+        eventProvider.getEventsWithPlaces(forLocation: location, usingPlacesDatabase: database) { places in
+            self.eventsPlaces[location] = places
+            self.displayPlaces(forLocation: location)
+        }
+    }
+
+    private func displayPlaces(forLocation location: CLLocation) {
+        guard let places = self.nearbyPlaces[location],
+            let eventPlaces = self.eventsPlaces[location] else { return }
+        let preparedPlaces = self.preparePlaces(places: union(ofNearbyPlaces: places, andEventPlaces: eventPlaces), forLocation: location)
```

Comment:
events are filtered before they get here. The filtering occurs in the `EventsController`